department-of-veterans-affairs/vets-website
# Pensions: replace deprecated components

## Summary

- Upgrades `childName`, `provider`, and `purpose` fields on the page `/financial/medical-expenses/add` in Pensions.

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/76778

## Testing done

- Updated unit tests
- Manual tests:
     - Use 'save in-progress menu' to open the app to `/financial/medical-expenses`
     - Say that you have expenses, then enter some expenses. At least one should be for the veteran's child. The page should behave as it did before the change.
     - Continue until the review page. Confirm that the medical expenses section appears as it did before the change, and can be edited.
     - Submit. Confirm in the Network tab that the correct medical expenses data was submitted.

## Screenshots

The section hasn't changed visually:
<img width="522" alt="Screenshot 2024-02-28 at 2 53 36 PM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/4e9ff2fa-8024-4b82-a0e1-706e27343ae0">
<img width="517" alt="Screenshot 2024-02-29 at 9 53 17 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/b9696435-deb4-457e-8f90-557cf8d88e90">

## What areas of the site does it impact?

Pensions Benefits

## Acceptance criteria

https://github.com/department-of-veterans-affairs/va.gov-team/issues/76778

### Quality Assurance & Testing


### Error Handling


### Authentication
